src/compiler/core.ts

```diff
@@ -2,6 +2,8 @@
 /// <reference path="performance.ts" />
 
 namespace ts {
+    // WARNING: The script `configureNightly.ts` uses a regexp to parse out these values.
+    // If changing the text in this section, be sure to test `configureNightly` too.

```

Comment:
@DanielRosenwasser @andy-ms 
The <a href="https://github.com/Microsoft/TypeScript/pull/16646#pullrequestreview-47396091">original justification</a> for the change was: to make version parsing easier.

Given how parsing issues escalated, while compiler gained unnecessary variable and all these troubling cross-referencing comments — is it better to revisit?

If TSC goes back to dumb simple `export const version = "2.5.0";` all these quirks go away. More to it, built TSC.JS has version in a plain parseable format.

Compare old simple way:
``` typescript
(function (ts) {
    ts.version = "2.4.1";
})(ts || (ts = {}));
```

With what we got today:
```typescript
(function (ts) {
    ts.versionMajorMinor = "2.5";
    ts.version = ts.versionMajorMinor + ".0-dev.20170719";
})(ts || (ts = {}));
```

All that trouble for the sake of splitting `versionMajorMinor` off `version`. Could have just gone `version.split('.').slice(0,2).join('.')`.